### PR TITLE
add attestation+dal support

### LINKED ISSUE #4: Add attestation+DAL support
1. Refer to [this](https://docs.google.com/document/d/184Dgs-zaTZv8bO03KOOHWwLHEMZtLo_OIfjo59u6yp8/edit#heading=h.fu4pbcp79zjg) for more details.
2. [x] New tag to be supported for Attestation+DAL but should maintain the watermark.

### PR DESCRIPTION
**Pull request by ajinkyaraj-23**
fixes #4

### CODE REVIEW

File: test/python/utils/message.py [hunk 1]
```diff
@@ -168,6 +169,46 @@ def raw(self) -> bytes:
...
+        self.slot = slot
+        self.op_level = op_level
+        self.op_round = op_round
+        self.block_payload_hash = scrub(block_payload_hash)
+        assert_data_size("block_payload_hash", self.block_payload_hash, 32)
+        self.dal_message = scrub(dal_message)
+
+    def raw(self) -> bytes:
+        raw = b''
+        raw += MagicByte.TENDERBAKE_ENDORSEMENT.to_bytes(1, byteorder='big')
```
Comment (spalmer25) [inline test/python/utils/message.py:202]: Are you sure that `TENDERBAKE_ENDORSEMENT` is the right magic byte ?

In `tezos/src/proto_alpha/lib_protocol/operation_repr.ml`:
```ocaml
let of_watermark = function
  | Signature.Custom b ->
      if Compare.Int.(Bytes.length b > 0) then
        match Bytes.get b 0 with
        | '\x12' ->
            Option.map
              (fun chain_id -> Preattestation chain_id)
              (Chain_id.of_bytes_opt (Bytes.sub b 1 (Bytes.length b - 1)))
        | '\x13' ->
            Option.map
              (fun chain_id -> Attestation chain_id)
              (Chain_id.of_bytes_opt (Bytes.sub b 1 (Bytes.length b - 1)))
        | '\x14' ->
            Option.map
              (fun chain_id -> Dal_attestation chain_id)
              (Chain_id.of_bytes_opt (Bytes.sub b 1 (Bytes.length b - 1)))
        | _ -> None
      else None
  | _ -> None
```
I'm not sure whether this function is related to our operations, but I think it's better to be sure.

File: test/python/utils/message.py [hunk 2]
```diff
@@ -168,6 +169,46 @@ def raw(self) -> bytes:
...
+        raw += self.chain_id.to_bytes(4, byteorder='big')
+        raw += self.branch
+        raw += OperationTag.ATTESTATION_WITH_DAL.to_bytes(1, byteorder='big')
+        raw += self.slot.to_bytes(2, byteorder='big')
+        raw += self.op_level.to_bytes(4, byteorder='big')
+        raw += self.op_round.to_bytes(4, byteorder='big')
+        raw += self.block_payload_hash
+        return raw
```
Comment (spalmer25) [inline test/python/utils/message.py:210]: The dal-endorsements have a `dal_attestation` field in addition to those for endorsements.

In `tezos/src/proto_alpha/lib_protocol/operation_repr.ml`:
```ocaml
  let consensus_content_encoding =
    obj4
      (req "slot" Slot_repr.encoding)
      (req "level" Raw_level_repr.encoding)
      (req "round" Round_repr.encoding)
      (req "block_payload_hash" Block_payload_hash.encoding)

  let dal_content_encoding =
    obj1 (req "dal_attestation" Dal_attestation_repr.encoding)

  let endorsement_encoding = consensus_content_encoding

  let endorsement_with_dal_encoding =
    merge_objs consensus_content_encoding dal_content_encoding
```

You may just have forgotten to add the `self.dal_message`.

File: test/python/utils/message.py (hunk 1 above)
Comment (spalmer25) [reply test/python/utils/message.py:202]: Oh no, I was looking at the Dal_attestation case, whereas the PR is about Attestation + Dal.
You can resolve it if you agree.

Comment (ajinkyaraj-23) [reply test/python/utils/message.py:202]: yeah attestation+DAL is seperate operation which has same magic byte and tag is 23.

File: test/python/utils/message.py (hunk 2 above)
Comment (ajinkyaraj-23) [reply test/python/utils/message.py]: Done

File: Makefile [hunk 3]
```diff
@@ -25,7 +25,7 @@ GIT_DESCRIBE ?= $(shell git describe --tags --abbrev=8 --always --long --dirty 2
 VERSION_TAG ?= $(shell echo "$(GIT_DESCRIBE)" | cut -f1 -d-)
 APPVERSION_M=2
 APPVERSION_N=4
-APPVERSION_P=6
+APPVERSION_P=7
```
Comment (spalmer25) [inline Makefile:28]: Why increase the version here?

Comment (ajinkyaraj-23) [reply Makefile:28]: For testing. Need to provide DAL testing to other teams so need to have separate version for clarity. So we can say version xxx supports DAL attestation.

**Review by spalmer25 — APPROVED** [review]

**Pull request merged**